adagrams/game.py

```diff
@@ -1,11 +1,151 @@
+import random
+
+POOL = {
+    "A":9,
+    "B":2,
+    "C":2,
+    "D":4,
+    "E":12,
+    "F":2,
+    "G":3,
+    "H":2,
+    "I":9,
+    "J":1,
+    "K":1,
+    "L":4,
+    "M":2,
+    "N":6,
+    "O":8,
+    "P":2,
+    "Q":1,
+    "R":6,
+    "S":4,
+    "T":6,
+    "U":4,
+    "V":2,
+    "W":2,
+    "X":1,
+    "Y":2,
+    "Z":1
+}
+
 def draw_letters():
-    pass
+    pool_list = []
+    chosen_letters = []
+
+    for key in POOL:
+        counter = POOL[key]
+        while counter > 0:
+            pool_list.append(key)
+            counter -= 1
+
+    while len(chosen_letters) < 10:
+        letter = random.choice(pool_list)
+        index = pool_list.index(letter)
+        pool_list.pop(index)
+        chosen_letters.append(letter)
+    
+    return chosen_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letters = []
+    word_cap = word.upper()
+
+    for i in range(0,len(word_cap)):
+        letters.append(word_cap[i])
+
+    count_if_false = 0
+
+    for letter in letters:
+        if letter not in letter_bank:
+            count_if_false += 1
+        elif letters.count(letter) > letter_bank.count(letter):
+            count_if_false += 1
+        else:
+            continue
+
+    if count_if_false == 0:
+        return True
+    else:
+        return False
 
 def score_word(word):
-    pass
+    score = 0
+    word = word.upper()
+    one_point_letters = ["A","E","I","O","U","L","N","R","S","T"]
+    two_point_letters = ["D","G"]
+    three_point_letters = ["B","C","M","P"]
+    four_point_letters = ["F","H","V","W","Y"]
+    five_point_letters = ["K"]
+    eight_point_letters = ["J","X"]
+    ten_point_letters = ["Q","Z"]
+
+    for letter in word:
+        if letter in one_point_letters:
+            score +=1
+        elif letter in two_point_letters:
+            score+=2
+        elif letter in three_point_letters:
+            score+=3
+        elif letter in four_point_letters:
+            score+=4
+        elif letter in five_point_letters:
+            score+=5
+        elif letter in eight_point_letters:
+            score+=8
+        else:
+            score += 10
```

Comment:
[comment] Checking each letter in a range like this is definitely more maintainable than checking for each letter one by one. It could be faster, though. Consider how many checks this code does if `letter` happens to be `"Z"`. How many more than `"A"`? If performance became a concern for this code, you could consider using a dictionary to map each possible letter to a score.

Building that dictionary is a little painful (and redundant), but we could imagine starting from a dictionary that uses a string of all the letters which share a score as the key, and the score as the value. That would be very maintainable and have no redundancy. Then we could write code to convert from that structure to a dictionary where each letter is its own key. This would let us have a maintainable data representation, AND quick score lookups!

That might look like starting with your variables containing lists and the if/else chain, and adding a function to build a per-letter dictionary of scores...

```python
def build_letter_to_score():
  """ produces a dictionary of letter to score, e.g. { "A": 1, "E": 1, ... "Z": 10 } """
  ...
```

You will often find that, on the job, this kind of extra effort is worth the development time to balance performance and maintainability.